src/main/java/uk/gov/register/serialization/CommandParser.java

```diff
@@ -30,7 +39,10 @@ public RegisterCommand newCommand(String s){
             case "add-item":
                 if (parts.length == 2) {
                     try {
-                        Item item = new Item(new ObjectMapper().readTree(parts[1]));
+                        String jsonContent = parts[1];
+                        canonicalJsonValidator.validateItemStringIsCanonicalized(jsonContent);
+                        String itemHash = DigestUtils.sha256Hex(jsonContent.getBytes());
+                        Item item = new Item(itemHash, new ObjectMapper().readTree(jsonContent));
```

Comment:
Can we add another method to `ObjectReconstructor`. This will just make sure that we are using consistent methods for the existing /load endpoint and the new one (only difference is we're not using `CanonicalJsonMapper` for the new one).

```
public JsonNode reconstruct(String[] jsonObjectsAsStrings) {
        return new ObjectMapper().readValue(e.getBytes(StandardCharsets.UTF_8), JsonNode.class));
    }
```